mesonbuild/compilers/compilers.py

```diff
@@ -871,12 +871,7 @@ def compile(self, code: 'mesonlib.FileOrString',
                 no_ccache = True
                 contents = code
             else:
-                srcname = code.fname
-                if not is_object(code.fname):
-                    with open(code.fname, encoding='utf-8') as f:
-                        contents = f.read()
-                else:
-                    contents = '<binary>'
+                contents = srcname = code.fname
```

Comment:
I'd prefer to also change the log message here to be more clear about what it's doing.

Reply:
done